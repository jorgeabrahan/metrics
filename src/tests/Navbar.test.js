import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { wrapRouter } from './testResources';

describe('Tests for <Navbar /> component', () => {
  render(wrapRouter());
  it('Should render the Home route by default', () => {
    expect(screen.getByText('List of available currencies')).not.toBeNull();
    expect(screen.getByText('Currencies')).not.toBeNull();
  });

  it('Shouldn\'t render the search bar by default', () => {
    expect(() => screen.getByRole('searchbox')).toThrow();
  });

  it('Should match snapshot', () => {
    expect(renderer.create(wrapRouter()).toJSON()).toMatchSnapshot();
  });
});
