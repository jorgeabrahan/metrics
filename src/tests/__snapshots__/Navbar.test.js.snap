// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Navbar /> component Should match snapshot 1`] = `
Array [
  <nav
    className="nav"
  >
    <h1
      className="nav__title"
    >
      Currencies
    </h1>
    <button
      className="reset"
      onClick={[Function]}
      type="button"
    >
      <span
        className="material-symbols-outlined"
      >
        search
      </span>
    </button>
  </nav>,
  <section>
    <div
      className="header"
    >
      <h2>
        Currency
        <br />
        Rates
      </h2>
      <small>
        Includes cryptocurrencies
      </small>
    </div>
    <small
      className="subtitle-separator"
    >
      List of available currencies
    </small>
    <div
      className="currencies"
    >
      <p>
        Loading...
      </p>
    </div>
  </section>,
]
`;
